suite-native/module-add-coin/src/hooks/useAddCoinAccount.ts

```diff
@@ -0,0 +1,250 @@
+    const [supportedNetworkSymbols, setSupportedNetworkSymbols] = useState<NetworkSymbol[]>([]);
+    const [addedAccount, setAddedAccount] = useState<Account | undefined>(undefined);
+    const [isAddingCoinAccount, setIsAddingCoinAccount] = useState<boolean>(false);
+
+    useEffect(() => {
+        const getSupportedCardanoTypes = async ({
+            deviceState,
+            trezorDevice,
+        }: {
+            deviceState: string;
+            trezorDevice: TrezorDevice;
+        }) => {
+            const types = await dispatch(
+                getCardanoSupportedAccountTypesThunk({
+                    deviceState,
+                    device: trezorDevice,
+                }),
+            ).unwrap();
+
+            if (types.success && types.payload) {
+                setAvailableCardanoTypes(types.payload ?? []);
+            } else {
+                setAvailableCardanoTypes([]);
+            }
+        };
+
+        if (device?.state !== undefined) {
+            // Cardano supported account types can vary based on device and firmware, so we need to fetch them
+            getSupportedCardanoTypes({ deviceState: device.state, trezorDevice: device });
+        } else {
+            setAvailableCardanoTypes([]);
+        }
+    }, [device, dispatch]);
+
+    useEffect(() => {
+        const symbols: NetworkSymbol[] = [];
+        supportedNetworks.forEach(element => {
+            if (symbols.indexOf(element.symbol) === -1) {
+                symbols.push(element.symbol);
+            }
+        });
+
+        setSupportedNetworkSymbols(symbols);
+    }, [supportedNetworks]);
+
+    const getAvailableAccountTypesForNetwork = useCallback(
+        ({ network }: { network: Network }) =>
+            network.networkType === 'cardano'
+                ? availableCardanoTypes
+                : availableNetworkAccountTypes.get(network.symbol),
+        [availableCardanoTypes],
+    );
+
+    const getDefaultAccountType = ({ network }: { network: Network }) =>
+        getAvailableAccountTypesForNetwork({ network })?.[0];
+
+    const getNetwork = useCallback(
+        ({
+            networkSymbol,
+            accountType,
+        }: {
+            networkSymbol: NetworkSymbol;
+            accountType?: AccountType;
+        }) => {
+            const type =
+                accountType ??
+                getDefaultAccountType({
+                    network: supportedNetworks.filter(
+                        network => network.symbol === networkSymbol,
+                    )[0],
+                });
+
+            return supportedNetworks.filter(
+                network =>
+                    network.symbol === networkSymbol &&
+                    (type === 'normal'
+                        ? network.accountType === undefined
+                        : network.accountType === type),
+            )[0];
+        },
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Seems like eslint is warning you about something here :D

Reply:
044d93041c